packages/babel-parser/src/plugins/jsx/index.js

```diff
@@ -614,14 +614,15 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     }
 
     updateContext(prevType: TokenType): void {
-      super.updateContext(prevType);
       const { context, type } = this.state;
       if (type === tt.braceL) {
         const curContext = context[context.length - 1];
         if (curContext === tc.j_oTag) {
           context.push(tc.brace);
         } else if (curContext === tc.j_expr) {
           context.push(tc.templateQuasi);
+        } else {
+          super.updateContext(prevType);
```

Comment:
This is incorrectly hoisted https://github.com/babel/babel/pull/13431/files#diff-064715ebec4ee229444ec2f30ab15e808d043e48dceb71baf6f579b35be31ba5R617